src/authn/index.js

```diff
@@ -153,7 +153,7 @@ function setup(app) {
   // requests to Cognito when we need to load back a user profile from their
   // session cookie.
   passport.serializeUser((user, done) => {
-    let serializedUser = {...user};
+    const serializedUser = {...user};
```

Comment:
> but I don't see any rationale.

The rationale is unstated, but I used `let` because the variables are modified later. The `let` signals to the reader that modification will happen.

The reason `const` works here is because it only prevents reassignment, not modification. So it's not constant in an intuitive nor useful sense when assigned an object or array (vs. a string/integer literal).

Reply:
Your proposal is to keep the original code:

```js
let serializedUser = {...user}; // eslint-disable-line prefer-const
```

How about using `const` to be "syntactically correct" along with a comment that more clearly signals modification to the reader? Example:

```js
// Note that this variable will be modified later.
const serializedUser = {...user};
```

Reply:
Hmm, I don't see how there's anything more "syntactically correct" about `const` than `let` here. It's a question of style since `const` isn't about actual immutability (and we're not dealing with memory layout implications, like in other languages).

As far as adding a comment goes, I think it's a little bit silly to have a comment saying "Hey, this thing we're calling a constant will, in fact, be modified below."  :upside_down_face: Seems less convoluted to just make it `let`.

Reply:
I didn't read up on differences and assumed there were some benefits to using `const`. Looking more, it seems like there isn't.

Replaced aed061754cc0c368c9affb9424f2a1df858a2da9 with f658bd7d43e10b78705c16a1a2b6e677b550c2a9.

Reply:
[Interested in thoughts -- I'm not trying to re-add this rule to this repo]

For booleans / numbers / strings diligently using `let/const` is useful (for me) when reading and writing code.
For objects (and all their derivatives), framing `let/const` in terms of immutability does make using them diligently seem somewhat pointless. However I've come to read it as something else, namely "this reference won't change"; which _is_ useful to me when reading code. It has particular relevance for things like React which use reference-based equality checks. My 2 cents 😄

Reply:
> However I've come to read it as something else, namely "this reference won't change"

I mean, that's literally what it means! so that's a reasonable reading of it. :-) I just think it's a not very useful choice of meaning and so prefer to either a) ignore the let/const distinction or b) assign `const` a meaning that I think is more useful (if not enforced by the language).

I'm not sure I understand how it's useful in relation to React. React begs for immutability but that's not what it gets with `const`. So if I see React code using a `const` variable and I read `const` as "this reference won't change" (it's literal meaning), I still have to think "well, is anything mutating the members of this object directly instead of passing React a modified copy?". When using `const` instead as "this variable won't change" (my preferred approach (b) above), then I know I don't have to think about what React will see because I wouldn't have marked it `const` if I was mutating it.